### DISCUSSION TITLE
Custom engine

### DISCUSSION THREAD

**Discussion by Collederas** [SmileyChris/easy-thumbnails#439] [Ideas]
Hi guys,

at the moment the official library to handle all images is _PIL_. You can, as stated in the docs, use other libraries by defining a custom _source_generator_.

I am trying to do so, because I need to use [_wand_](http://docs.wand-py.org/en/0.4.4/), a python wrapper around [_imagemagick_](http://www.imagemagick.org/script/index.php).

Everything works fine except the fact that all the methods in `engine.py` are tailored to _PIL_. In particular the `save_image()` method calls `image.save()` with a lot of arguments that are not existent in, for example, _wand_.

My question then is: if the _source_generator_ is customizable in the settings, why not making customizable also the _engine_?

Thanks for the attention!
Best,

Fabrizio

**Comment by SmileyChris** [comment]
Hi Fabrizio, you should check out my https://github.com/SmileyChris/easy-images alpha and see if the ideas there would be easier to work with. I'd love some feedback :)
